### DISCUSSION TITLE
Skin elements for current player information

### DISCUSSION THREAD

**Discussion by Bruno5430** [ppy/osu#22761] [Ideas]
(Just a proposal, not going to implement this)

As i saw more and more skins where people embed their avatars and usernames in the texture of the health bar, i was thinking if it could be done in a more procedural way with skin elements from the skin editor

![playerinfoconcept](https://user-images.githubusercontent.com/111712756/222211801-0043fc6b-39af-433e-9ba3-11c9b91ec6a1.png)

In legacy skins with embedded avatars on the health bar will always show the same avatar and username
![2023-03-01_14-20](https://user-images.githubusercontent.com/111712756/222223575-e83f5bb5-4bb1-4339-9057-9d9953f691b7.png)
but if they were skin elements, then they could change not only for the logged in player, but also could display a player from a replay, or even for multiple players while spectating a multiplayer room

I propose 3 new elements
`PlayerAvatar`, `PlayerFlag` and `PlayerAttributeText`

`PlayerAvatar` is very self-explanatory.
An optional request would be to have a slider in the options of the element to customize the radius of the round borders (Setting the slider at maximum value would make the element completely round like github avatars). If not, make the radius match osu-web
![2023-03-01_14-52](https://user-images.githubusercontent.com/111712756/222221805-efc77391-1db5-457c-97df-10a04058578e.png)

`PlayerFlag` would display the country flag of the player (Not the name of the country)

`PlayerAttributeText` would follow the behavoiur of `BeatmapAttributeText` but for attributes of the player, like their username, global rank on the current ruleset, national rank, name of their country, and **maybe** other miscellaneous statistics like level, total pp, hit accuracy, max combo, etc. (not sure of the naming of the values)

Edit: i forgot to mention the username on `PlayerAttributeText`

**Comment by ItsShamed** [comment]
I am absolutely down for having a `PlayerAvatar` and a `PlayerFlag` skinnable component. However, I'm not really fond of having another `AttributeText` thing, simply because of how things are structured right now (I do not agree with the current structure of the text components).

**Comment by peppy** [comment]
Seems like a good thing to have available, for sure.

**Reply by peppy** [reply]
Partially completed via #22797
